agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -725,55 +717,81 @@ static Configuration getConfigurationFromEnvVar(String content) {
+    }
+    String content = new String(bytes, StandardCharsets.UTF_8);
+    return getConfiguration(content, strict, ConfigurationOrigin.fromPath(configPath));
+  }
+
   private static Configuration getConfiguration(
       String content, boolean strict, ConfigurationOrigin configurationOrigin) {
-    Configuration configuration;
+    configurationLogger.debug("configuration content: {}", content);
     ObjectMapper mapper = new ObjectMapper();
     if (!strict) {
       mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     }
+    JsonNode jsonNode;
+    try {
+      jsonNode = mapper.readTree(content);
+    } catch (JsonProcessingException e) {
+      throw new FriendlyException(
+          "The configuration "
+              + configurationOrigin
+              + " contains malformed JSON."
+              + System.lineSeparator()
+              + System.lineSeparator()
+              + e.getMessage(),
+          "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358",
+          e);
+    }
     try {
-      configuration = mapper.readValue(content, Configuration.class);
-    } catch (UnrecognizedPropertyException ex) {
+      return mapper.treeToValue(jsonNode, Configuration.class);
+    } catch (UnrecognizedPropertyException e) {
       if (strict) {
-        // Try extracting the configuration without failOnUnknown
-        configuration = getConfiguration(content, false, configurationOrigin);
-        // cannot use logger before loading configuration, so need to store warning messages locally
-        // until logger is initialized
+        Configuration configuration = getConfiguration(content, false, configurationOrigin);
         configurationLogger.warn(
-            getJsonEncodingExceptionMessage(ex.getMessage(), configurationOrigin));
+            getJsonEncodingExceptionMessage(e.getMessage(), configurationOrigin), e);
+        return configuration;
       } else {
         throw new FriendlyException(
-            getJsonEncodingExceptionMessage(ex.getMessage(), configurationOrigin),
+            getJsonEncodingExceptionMessage(e.getMessage(), configurationOrigin),
             "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
       }
-    } catch (JsonMappingException | JsonParseException ex) {
+    } catch (JsonParseException | JsonMappingException e) {
       throw new FriendlyException(
-          getJsonEncodingExceptionMessage(ex.getMessage(), configurationOrigin),
-          "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
+          "Error parsing configuration from "
+              + configurationOrigin
+              + "."
+              + System.lineSeparator()
+              + System.lineSeparator()
+              + e.getMessage(),
+          "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358",
```

Comment:
`https://go.microsoft.com/fwlink/?linkid=2153358` appears at least twice in the code, if this URL changes, the code maintenance would be easier with a constant